File: app/arm/src/main/java/tech/yaowen/arm/MainActivity.java
package tech.yaowen.arm;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

public class MainActivity extends AppCompatActivity {
    static {
        System.loadLibrary("arm");
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Object o = new Object();
        testHWASan();
    }

    native void testCAS();

    native void testHWASan();
}

File: test_glide/src/main/java/tech/yaowen/test_glide/MainActivity.kt
package tech.yaowen.test_glide

import android.os.Bundle
import android.view.View
import android.widget.ImageView
import androidx.annotation.NonNull
import androidx.appcompat.app.AppCompatActivity
import com.bumptech.glide.Glide
import com.bumptech.glide.RequestBuilder
import com.bumptech.glide.annotation.GlideType
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions
import com.bumptech.glide.load.resource.gif.GifDrawable
import com.bumptech.glide.request.RequestOptions
import com.bumptech.glide.request.RequestOptions.decodeTypeOf


class MainActivity : AppCompatActivity() {
    lateinit var imageView: ImageView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        imageView = findViewById(R.id.imageView)
    }

    fun loadImage(view: View) {
        Glide.with(view)
//            .asMyGif()
            .load("https://p3.pstatp.com/large/37db000436571162d73f")
            
            .into(imageView)
    }

}
